lib/os/thread_posix.cpp

```diff
@@ -70,29 +66,34 @@ namespace pstore {
 #        elif defined(PSTORE_PTHREAD_SET_NAME_NP)
             pthread_set_name_np (pthread_self (), name);
 #        else
-#            error "pthread_setname was not available"
+            std::strncpy(thread_name, name, name_size);
+            thread_name[name_size - 1] = '\0';
 #        endif
             if (err != 0) {
                 raise (errno_erc{err}, "pthread_set_name_np");
             }
-#    endif // __FreeBSD__
         }
 
         gsl::czstring get_name (gsl::span<char, name_size> const name /*out*/) {
             auto const length = name.size ();
             if (name.data () == nullptr || length < 1) {
                 raise (errno_erc{EINVAL});
             }
-#    ifdef __FreeBSD__
-            std::strncpy (name.data (), thread_name, static_cast<std::size_t> (length));
-#    else
+
             PSTORE_ASSERT (length == name.size_bytes ());
-            int const err = pthread_getname_np (pthread_self (), name.data (),
-                                                static_cast<std::size_t> (length));
+            int err = 0;
+#        if defined(PSTORE_PTHREAD_GETNAME_NP)
+            err = pthread_getname_np (pthread_self (), name.data (),
+                                      static_cast<std::size_t> (length));
+#        elif defined(PSTORE_PTHREAD_GET_NAME_NP)
+            err = pthread_get_name_np (pthread_self (), name.data (),
+                                       static_cast<std::size_t> (length));
+#        else
+            std::strncpy(name.data(), thread_name, static_cast<std::size_t> (length));
```

Comment:
See above.

Reply:
Thanks, fixed in the commit https://github.com/SNSystems/pstore/pull/93/commits/88195a24bcbd4e6e349f08bc63cdc1197b10f153.